alws/models.py

```diff
@@ -4,15 +4,87 @@
 import re
 
 import sqlalchemy
-from sqlalchemy.orm import relationship
+from sqlalchemy.orm import relationship, declared_attr, declarative_mixin
 from sqlalchemy.dialects.postgresql import JSONB
 from sqlalchemy.ext.associationproxy import association_proxy
 
-from alws.constants import ReleaseStatus, SignStatus
+from alws.constants import (
+    Permissions,
+    PermissionTriad,
+    ReleaseStatus,
+    SignStatus,
+)
 from alws.database import Base, engine
 
 
-__all__ = ['Platform', 'Build', 'BuildTask', 'Distribution']
+__all__ = [
+    'Build',
+    'BuildTask',
+    'Distribution',
+    'Platform',
+    'SignKey',
+    'SignTask',
+    'User',
+    'UserAction',
+    'UserRole',
+    'Team',
+]
+
+
+@declarative_mixin
+class TeamMixin:
+
+    @declared_attr
+    def team_id(cls):
+        # FIXME: Change nullable to False after owner population
+        return sqlalchemy.Column(
+            sqlalchemy.Integer, sqlalchemy.ForeignKey(
+                'teams.id', name=f'{cls.__tablename__}_team_id_fkey'),
+            nullable=True)
+
+    @declared_attr
+    def team(cls):
+        return relationship('Team')
+
+
+@declarative_mixin
+class PermissionsMixin:
+
+    @declared_attr
+    def owner_id(cls):
+        # FIXME: Change nullable to False after owner population
+        return sqlalchemy.Column(
+            sqlalchemy.Integer, sqlalchemy.ForeignKey(
+                'users.id', name=f'{cls.__tablename__}_owner_id_fkey'),
+            nullable=True)
+
+    @declared_attr
+    def owner(cls):
+        return relationship('User')
+
+    permissions = sqlalchemy.Column(sqlalchemy.Integer, nullable=False,
+                                    default=764)
+
+    @property
+    def permissions_triad(self):
+        self.validate_permissions(self.permissions)
+        return PermissionTriad(
+            Permissions(self.permissions // 100),
+            Permissions(self.permissions // 10 % 10),
+            Permissions(self.permissions % 10),
```

Comment:
We already has all of the values in `validate_permissions`, may be we could return tuple `(owner, group, other)` from here? Because this code is a little obfuscated